static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx

```diff
@@ -0,0 +1,109 @@
+import {useEffect} from 'react';
+import styled from '@emotion/styled';
+
+import {openModal} from 'sentry/actionCreators/modal';
+import {Button} from 'sentry/components/button';
+import {Tooltip} from 'sentry/components/tooltip';
+import {t} from 'sentry/locale';
+import PluginIcon from 'sentry/plugins/components/pluginIcon';
+import {space} from 'sentry/styles/space';
+import type {Organization} from 'sentry/types/organization';
+import type {Project} from 'sentry/types/project';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import MessagingIntegrationModal from 'sentry/views/alerts/rules/issue/messagingIntegrationModal';
+
+type Props = {
+  organization: Organization;
+  projectSlug: string;
+  refetchConfigs: () => void;
+};
+
+function SetupMessagingIntegrationButton({
+  organization,
+  projectSlug,
+  refetchConfigs,
+}: Props) {
+  const providerKeys = ['slack', 'discord', 'msteams'];
+
+  const onAddIntegration = () => {
+    refetch();
+    refetchConfigs();
+  };
+
+  const {
+    data: project,
+    isLoading,
+    isError,
+    refetch,
+  } = useApiQuery<{}>(
+    [`/projects/${organization.slug}/${projectSlug}/?expand=hasAlertIntegration`],
+    {staleTime: Infinity}
+  );
+
+  const detailedProject = project as Project & {
```

Comment:
You can define this type outside the component as:


```ts
interface ProjectWithAlertIntegrationInfo extends Project {
  hasAlertIntegrationInstalled: boolean;
}
```

Using `interface` and `extends` is better when adding properties to an existing type since it is more performant and it works more how you would expect it to (type intersection can result in some confusing types in some situations)